modules/aws/src/smithy4s/aws/AwsCredentialsFile.scala

```diff
@@ -0,0 +1,158 @@
+ *  limitations under the License.
+ */
+
+package smithy4s.aws
+
+import cats.effect.Resource
+import cats.effect.Sync
+import cats.syntax.all._
+import java.nio.file.Paths
+import smithy4s.aws.kernel.AWS_ACCESS_KEY_ID
+import smithy4s.aws.kernel.AWS_SECRET_ACCESS_KEY
+import smithy4s.aws.kernel.AWS_SESSION_TOKEN
+import smithy4s.aws.kernel.AwsCredentials
+
+final case class AwsCredentialsFile(
+    default: Option[AwsCredentials],
+    profiles: Map[String, AwsCredentials]
+)
+
+object AwsCredentialsFile {
+  def loadFromDisk[F[_]](home: String, profile: Option[String])(implicit
+      F: Sync[F]
+  ): F[AwsCredentials] = {
+    loadFile(home)
+      .use(lines => processFileLines(lines).pure[F])
+      .flatMap(creds =>
+        profile
+          .map(p =>
+            creds.profiles
+              .get(p)
+              .toRight(
+                new RuntimeException(
+                  s"Profile `$p` was not found in the credentials file."
+                )
+              )
+              .liftTo[F]
+          )
+          .getOrElse(
+            creds.default
+              .toRight(
+                new RuntimeException(
+                  s"No default profile is available in the credentials file."
+                )
+              )
+              .liftTo[F]
+          )
+      )
+  }
+
+  private[aws] def processFileLines(lines: List[String]): AwsCredentialsFile = {
+    def inProfile(
+        rest: List[String],
+        currentProfile: String,
+        data: Map[String, String]
+    ): (Map[String, String], List[String]) = {
+      rest match {
+        case Nil =>
+          (data, Nil)
+        case head :: _ if head.trim().startsWith("[") =>
+          (data, rest)
+        case head :: next =>
+          val parts = head.split("=")
+          val key = parts(0).trim().toLowerCase
+          val value = parts(1).trim().toLowerCase()
+          val updated = data + (key -> value)
+          inProfile(next, currentProfile, updated)
+      }
+    }
+
+    def lookingForProfile(
+        rest: List[String],
+        data: Map[String, Map[String, String]]
+    ): Map[String, Map[String, String]] = {
+      rest match {
+        case head :: next =>
+          val profile = head.trim() match {
+            case Profile.Prefixed(profile)    => profile
+            case Profile.Default(profile)     => profile
+            case Profile.NonPrefixed(profile) => profile
+          }
+
+          val (profileData, rest) = inProfile(next, profile, Map.empty)
+          lookingForProfile(rest, data + (profile -> profileData))
+        case Nil =>
+          data
+      }
+    }
+
+    val data = lookingForProfile(lines.filter(_.trim.nonEmpty), Map.empty)
+    AwsCredentialsFile.profilesFromMap(data)
+  }
+
+  private def loadFile[F[_]: Sync](home: String): Resource[F, List[String]] =
+    Resource
+      .fromAutoCloseable(
+        Sync[F].delay(
+          scala.io.Source
```

Comment:
Unfortunately, this is not gonna be working on node environments. Let's use [fs2.io](https://fs2.io/#/io?id=files) instead, that'll work on the three environments.